### PR TITLE
History page + store wallet type

### PR DESCRIPTION
**Pull request by skogman**
Add history page and store wallet type in store from local storage
screen shot 2562-03-01 at 18 36 52

### CODE REVIEW

File: src/components/Layout.vue [hunk 1]
```diff
@@ -187,7 +188,7 @@ export default class Layout extends Vue {
     
     if(window.ethereum) {
       window.ethereum.on('accountsChanged', async (accounts) => {
```
Comment (lukezhangstudio) [inline src/components/Layout.vue]: on accounts changed, we should just have it log the user out

File: src/components/Layout.vue [hunk 2]
```diff
@@ -132,9 +133,9 @@ export default class Layout extends Vue {
   }
 
   beforeMount() {
-    if(localStorage.getItem("privatekey")) {
-      this.setUserIsLoggedIn(true)
-    }
+    if(localStorage.getItem("privatekey")) this.setUserIsLoggedIn(true)
```
Comment (lukezhangstudio) [inline src/components/Layout.vue]: i changed this and now if it detects private key, it just logs the user out

File: src/store/dposStore.js [hunk 3]
```diff
@@ -51,15 +51,24 @@ const defaultState = () => {
       // { key: 'Uptime', sortable: true },
       // { key: 'Slashes', sortable: true },
     ],
-    prohibitedNodes: ["plasma-0", "plasma-1", "plasma-2", "plasma-3", "plasma-4", "Validator #4", "test-z-us1-dappchains-2-aws0"]
+    prohibitedNodes: ["plasma-0", "plasma-1", "plasma-2", "plasma-3", "plasma-4", "Validator #4", "test-z-us1-dappchains-2-aws0"],
+    latestBlockNumber: 0,
```
Comment (lukezhangstudio) [inline src/store/dposStore.js]: jesus, we shouldn't start from block 0, make it start from last 10,000 blocks or something

File: src/views/History.vue [hunk 4]
```diff
@@ -1,41 +1,185 @@
...
-        <router-link to="/browse" class="current">{{ $t('views.history.trade_history') }}</router-link>
+  <div class="faucet">
+    <div class="faucet-content">
+      <div class="container mb-5 column py-3 p-3 d-flex">
+        <h1>{{ $t('views.history.trade_history') }}</h1>
+        <div class="py-3">
+          <div v-if="showHistoryTable"><faucet-table :items="history"></faucet-table></div>
+          <div v-else>
+            <h5>
+              You haven't made any trades yet
```
Comment (lukezhangstudio) [inline src/views/History.vue]: should say no activity detected in last x number of blocks, and make a link to go back further in time

**Comment by lukezhangstudio** [comment]
we should also change the title from 'trade history' to "your history". should also plan on adding a filter above the table, since we'll have: ethereum events, dappchain gateway events, and delegation events.

File: src/components/Layout.vue [hunk 5]
```diff
@@ -132,10 +133,7 @@ export default class Layout extends Vue {
   }
 
   beforeMount() {
-    if(localStorage.getItem("privatekey")) {
-      // this.setUserIsLoggedIn(true)
-      this.$router.push({ path: '/login' })
-    }
+    if(!this.userIsLoggedIn) this.$router.push({ path: '/login' })
```
Comment (lukezhangstudio) [inline src/components/Layout.vue]: sorry i don't understand this change. we want to make sure to log people out if they're logged in, and not the opposite

File: src/components/Layout.vue [hunk 6]
```diff
@@ -188,7 +186,7 @@ export default class Layout extends Vue {
     
     if(window.ethereum) {
       window.ethereum.on('accountsChanged', async (accounts) => {
-        if(this.userIsLoggedIn) this.ensureIdentityMappingExists({currentAddress: accounts[0]})
+        if(this.userIsLoggedIn && this.walletType === "metamask") this.ensureIdentityMappingExists({currentAddress: accounts[0]})
```
Comment (lukezhangstudio) [inline src/components/Layout.vue]: what's the reason for this change?

File: src/components/Layout.vue (hunk 5 above)
Comment (skogman) [reply src/components/Layout.vue]: Hmm, my reasoning is, If the user refreshes the page `this.userIsLoggedIn` will be false, in which case we want to redirect them to the `/login` page?

File: src/components/Layout.vue (hunk 6 above)
Comment (skogman) [reply src/components/Layout.vue]: This is a mistake on my part, I've updated the code.

File: src/components/Layout.vue (hunk 5 above)
Comment (lukezhangstudio) [reply src/components/Layout.vue]: k, i'll merge and then we test from there
